EIPS/eip-1238.md

```diff
@@ -0,0 +1,213 @@
+discussions-to: https://ethereum-magicians.org/t/eip-1238-non-transferable-tokens/9044
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-07
+requires: 165, 712
+---
+
+
+
+## Abstract
+A _badge_ or non-transferable token (_NTT_) is a token that cannot be transferred once assigned. Badges can be accumulated through time and put at stake. Simply speaking, badges are statements about a public key: they can be quantitative (e.g. reputation, experience) or qualitative (badges, titles).
+
+The Non-Transferable Token standard defines a set of standard APIs allowing the identification of statements (called badges) attributed to a public key, such that different dapps and smart contract can use to filter users or to provide user with different badges different experiences. More importantly, this standard defines a way for which users can put their badges at stake. Badges cannot be transferred but can be lost (after staking) or can expire.
+
+## Motivation
+The idea is to have tokens that once assigned cannot be transferred (like reputation) and that can be used by websites, or contracts to make users perform some actions. For example, if a user accumulates paper submissions at conferences, then they can use their paper badges to request grants. It's important that they can never share these badges.
+
+This is the equivalent of a variety of other use cases
+
+- Accumulating badges: degrees from academic institutions, paper publications for conferences, badges that allow for access in a building or in a special chat.
+- Experience points: points accumulated in a game, points accumulated by being honest in a decentralized system for some time or a DAO.
+- Statements: more broadly, any statement that is issued or signed by a contract, a dao, a single user that requires to be on-chain.
+- Subscription: badges can represent the validity of a paid subscription.
+
+Additionally, this standard could be used as a primitive for Soulbound tokens, whereby the recovery of a Soul would be done by obtaining signatures from co-owners of tokens with the same id.
+
+Multiple proposals for non-transferable / Soulbound tokens have been put forth, however this one presents a unique combination of features which includes having multiple token types in one contract (similar to [EIP-1155](./eip-1155.md)), recipient's consent and, optionally, token staking and expiry.
+
+## Specification
+Every contract compliant with this EIP must implement the following interface:
+
+```solidity
+interface IERC1238 is IERC165 {
+    /**
+     * @dev Emitted when `amount` tokens of token type `id` are minted to `to` by `minter`.
+     */
+    event MintSingle(address indexed minter, address indexed to, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Equivalent to multiple {MintSingle} events, where `minter` and `to` is the same for all token types.
+     */
+    event MintBatch(address indexed minter, address indexed to, uint256[] ids, uint256[] amounts);
```

Comment:
Why do you have both `MintSingle` and `MintBatch`? Couldn't `MintBatch` handle both cases?

If there's a reason, please add it to the EIP's rationale section.

Reply:
Good point, I like how it makes the interface simpler. I also applied the same reasoning to the events related to burning.